### PR TITLE
Fix bug in calibration that results in empty Sv

### PR DESCRIPTION
**Pull request by leewujung**
The renaming of time coordinate in the `Environment` group (now called `time1` according to the convention) results in discrepancy in broadcasting the environment parameters during calibration, since the backscatter data is aligned with `ping_time`. In v0.6.0 `time1` was directly renamed to `ping_time`, which resulted in all non-matching timestamps in the backscatter data to be broadcasted to NaN. 

I think this is related to some parts of the problems seen in a number of recent issues #720 #743 #750. 

This is high priority and let's get the fix out in v0.6.1.

### Tasks
- [x] fix bugs related to time coordinate broadcasting
- [ ] add unit tests for the new function `EchoData._harmonize_env_param_time`
- [ ] add tests for catching unexpected NaN in calibrated data

### CODE REVIEW

**Comment by leewujung** [comment]
Oops circular import. I'll work on that.

File: echopype/echodata/echodata.py [hunk 1]
```diff
@@ -256,6 +256,42 @@ def __setattr__(self, __name: str, __value: Any) -> None:
...
 
+    def _harmonize_env_param_time(
+        self, p: Union[int, float, xr.DataArray], ping_time: Union[xr.DataArray, datetime.datetime]
+    ):
+        """
+        Harmonize time coordinate between Beam_groupX data and env_params to make sure
+        the timestamps are broacast correctly in calibration and range calculations.
+
+        If timestamp exist for env_params (i.e. it is of type xr.DataArray),
+        they cpme from EchoData["Environment"] with coordinate time1.
```
Comment (emiliom) [inline echopype/echodata/echodata.py:267]:
```suggestion
        If timestamps exist for env_params (i.e. it is of type xr.DataArray),
        they come from EchoData["Environment"] with coordinate time1.
```

File: echopype/echodata/echodata.py [hunk 2]
```diff
@@ -256,6 +256,42 @@ def __setattr__(self, __name: str, __value: Any) -> None:
...
+        For EK60, Environment.time1 = Beam_group1.ping_time, both from the RAW0 datagram
+        so are directly interchangeable.
+        For EK80, Environment.time1 is from the Environment XML datagram,
```
Comment (emiliom) [inline echopype/echodata/echodata.py:270]: I assume that's true also for AZFP? Plus, that's how you've implemented it, below. 

How about ES70, ES80 and EA640?

File: echopype/echodata/echodata.py [hunk 3]
```diff
@@ -256,6 +256,42 @@ def __setattr__(self, __name: str, __value: Any) -> None:
                         self._tree[group_path].ds = __value
         super().__setattr__(__name, attr_value)
 
+    def _harmonize_env_param_time(
+        self, p: Union[int, float, xr.DataArray], ping_time: Union[xr.DataArray, datetime.datetime]
```
Comment (emiliom) [inline echopype/echodata/echodata.py:260]:
```suggestion
        self, p: Union[int, float, xr.DataArray], ping_time: Union[xr.DataArray, datetime.datetime] = None
```

This change shouldn't have any effect, but it'll make it clearer that `ping_time` is in effect optional. It's not used (and therefore not passed) for EK60. Maybe you can add a test for `ping_time is not None` below, before the statement `p_new = p.interp(time1=ping_time)`

Comment (leewujung) [reply echopype/echodata/echodata.py:260]: Yes, thanks for catching this. I did the typing and forgot to add back the =None part.

And good point for testing for `ping_time` -- this interp part is not used at all in the current code (since our time1 is always length=1 from EK80), but it will surely come in later once we combine echodata. I will add a test for this too, to make sure the operations are done correctly (although it is arguable whether we should do `method="linear"` or `method="nearest"` or something else).

File: echopype/echodata/echodata.py (hunk 2 above)
Comment (leewujung) [reply echopype/echodata/echodata.py:270]: ES70 in theory has the same format as EK60 but I have not actually handle an example file. ES80 and EA640 has the same format as EK80. I think we have ES80 and EA640 test files, and can try to run them through the calibration code to see what happens.

File: echopype/echodata/echodata.py (hunk 3 above)
Comment (leewujung) [reply echopype/echodata/echodata.py]: I decided to use `method="linear"` and remove the check for `sonar_model` so that the function is general. I realized that even for EK60 it could be that users provide data arrays as inputs of the `env_params`, and in that case we can't just do the `time1` to `ping_time` renaming either.

File: echopype/echodata/echodata.py (hunk 2 above)
Comment (leewujung) [reply echopype/echodata/echodata.py]: I've updated the function so that `sonar_model` is no longer used.

File: echopype/echodata/echodata.py (hunk 3 above)
Comment (leewujung) [reply echopype/echodata/echodata.py]: Done!
